### DISCUSSION TITLE
Production deployment

### DISCUSSION THREAD

**Discussion by jpbalmes** [Tecnativa/doodba#484] [Q&A] [answered]
Hello everyone!

I am beginning to take my first steps with Doodba and I have a question at the time of deployment in production.

With the initial deploy I had no problems. Following the steps in the guide (https://github.com/Tecnativa/doodba-copier-template/blob/main/docs/daily-usage.md#booting-production), I ran "docker-compose -f prod.yaml up -dy Odoo got up without problems on my VPS.

The doubt arises when I want to install apps/modules that I incorporated into the repos.yaml and addons.yaml files after the initial deploy.
The dev environment does not generate a problem for me, I execute these commands and the new modules are available to be installed:

    docker-compose build --pull
    docker-compose -f setup-devel.yaml up
    docker-compose up

**Then, in the production environment, with the modified repos.yaml and addons.yaml files (with the new repos and modules), what commands should I run to detect and download the new code?**
If I run the command "docker-compose -f prod.yaml up -d" again, I see that the docker image remains unchanged and it also doesn't run the git-aggregator again.

I am using the Tecnativa template: https://github.com/Tecnativa/doodba-copier-template/ with an Odoo v15.

Thank you very much in advance.

**Comment by pedrobaeza** [comment, marked as answer]
`docker-compose build --pull`

**Reply by jpbalmes** [reply]
@pedrobaeza thanks for your answer. she forgot to mention it. That command was also executed but did not achieve the expected result. I attach a screenshot of the result
![Selección_178](https://user-images.githubusercontent.com/92761956/156626706-bcd52019-6276-40ad-878e-9d2d4afd9160.png)
.

**Reply by pedrobaeza** [reply]
Doing that + `docker-compose up -d` should be enough for getting the new instance with such addons.

**Reply by HviorForgeFlow** [reply]
If you push new commits to a branch that has already been referenced on repos.yaml, you must use 
```docker-compose build --pull --no-cache``` It will take longer to build the container but new changes will be added.

**Reply by juanpgarza** [reply]
@HviorForgeFlow OK, I'm pretty sure that's what caused my initial problem. I'll keep it in mind. Thank you very much!
